### ISSUE TITLE
Investigate options for dbt failure notifications

### ISSUE THREAD

**Issue by jkarpen** [cagov/data-infrastructure#415]

**Comment by summer-mothwood** [comment]
Looking more into dbt Cloud's webhook capabilities (and reviewing @ram-kishore-odi 's [write up](https://docs.google.com/document/d/1K8L-lci-0AjuPzbviANYS6lTXgtzUY7QEsAfKeuupps/edit?usp=sharing) for Andrew on a somewhat similar issue) -- it looks like Microsoft Teams has some built-in functionality for subscribing to webhooks, too, so we might not even need the Zapier intermediate step!

**Comment by britt-allen** [comment]
We could send an email to a channel if we were not on the 365 Gov't plan... ![Image](https://github.com/user-attachments/assets/b36eeeb5-89e7-4b9c-99e7-fdc50659c40b)

**Comment by britt-allen** [comment]
[This option](https://learn.microsoft.com/en-us/azure/data-factory/how-to-send-notifications-to-teams?tabs=data-factory) could work, but I cannot install the required app because I had to make a request which I imagine goes to an Admin at GovOps. Lots of our ability to explore is blocked by ODI not having its own Microsoft license yet.

**Comment by britt-allen** [comment]
I have checked off Teams as not viable for now.

**Comment by summer-mothwood** [comment]
I'm looking into setting up Slack notification in dbt cloud. The first step is to set up Slack as an Account Integration in dbt Cloud. When I try to click through here
![Image](https://github.com/user-attachments/assets/3f3bda47-2004-4b7e-a1b9-20b84bbdaa54)

I get this error
![Image](https://github.com/user-attachments/assets/5b32c34a-ab14-4fad-8466-a425b691ce0a)

Looking into it further (using these instructions https://docs.getdbt.com/docs/deploy/job-notifications#slack-notifications), I notice the docs say 

> You must be an account admin to configure Slack notifications in dbt Cloud. 

I tried changing my license to an IT license, but still got the same error. @britt-allen do you know if one of us is an account admin on the dbt cloud account that has elevated privileges for that kind of thing? or should I put in a request ticket with dbt?

**Comment by britt-allen** [comment]
Let me give it a try!

**Comment by britt-allen** [comment]
I get the same error as well. I think this is because how IT has setup our dbt account, I'd go to our IT team first before opening a ticket with dbt @summer-mothwood

**Comment by summer-mothwood** [comment]
Thanks for looking into it @britt-allen !! I'll reach out to IT -- we'll need a system admin on the Slack side to complete the next steps anyway, so I'll ask them about that while I'm at it too

**Comment by summer-mothwood** [comment]
Put in a ticket to IT to integrate Slack and dbt for us.

**Comment by summer-mothwood** [comment]
I met with Kevin in IT today, and he was unable to configure Slack notifications / add dbt as an integrated app into Slack either. He's going to follow up with Chad and will notify me (in Slack, hah!) with updates. I mentioned again that our preference would be to enable notifications in Teams in order to make it easy for our clients who don't use Slack. Keeping the status here as blocked.

**Comment by summer-mothwood** [comment]
Chad ran into the same issue, so Kevin filed a ticket with dbt. Response from their team here, Kevin is working on implementing this morning:

> Thank you for your patience. So it appears that there is an additional step that needs to be taken for Slack integration given your account is hosted on a Single Tenant instance. Would you be able to kindly do the following:
>  
> Create a Slack application here: [https://api.slack.com/apps](https://www.google.com/url?q=https://api.slack.com/apps&source=gmail-imap&ust=1732148394000000&usg=AOvVaw3FBW1ljudDZoeYOeP_MtSH)
> Once the above is completed, kindly provide the client ID and client secret. Please ensure to send the Client ID in plaintext and provide the Client secret using a self destructing service such as [https://1ty.me/](https://www.google.com/url?q=https://1ty.me/&source=gmail-imap&ust=1732148394000000&usg=AOvVaw24wxP3X0XG_cq1E1-LcR_3) for security purposes.
>  
> We're also looking into adding the additional requirement to our documentation here: [https://docs.getdbt.com/docs/deploy/job-notifications#slack-notifications](https://www.google.com/url?q=https://docs.getdbt.com/docs/deploy/job-notifications%23slack-notifications&source=gmail-imap&ust=1732148394000000&usg=AOvVaw00tvJy2z5vqK7MlOj8TQah).

**Comment by summer-mothwood** [comment]
Meeting scheduled with Chris Saur 12/10

**Comment by jkarpen** [comment]
@summer-mothwood I created this issue for the Listserv: https://github.com/cagov/caldata-dse-internal-tracker/issues/57

@ian-r-rose are you already creating issues for setting up Fivetran and other accounts for HFA? If not I can start those.

**Comment by ian-r-rose** [comment]
I have not yet created those issues, please do!

Ian Rose (he/him)
Principal Data Engineer, Data Services | CA Office of Data and Innovation
(916) 931-8386 | ***@***.***
Follow us: web <https://innovation.ca.gov/> | blog
<https://innovation.ca.gov/blog/> | twitter
<https://twitter.com/californiaodi>

**Comment by jkarpen** [comment]
Closing this issue as we have investigated and found a path forward. Might create a new issue for tracking the follow-on task of getting the dbt/slack integration set up but that is more on the IT side.
